crates/oxc_linter/src/rules/oxc/no_map_spread.rs

```diff
@@ -0,0 +1,578 @@
+use oxc_semantic::{Reference, ReferenceId, ScopeId, SymbolId};
+use serde::{Deserialize, Serialize};
+use std::ops::Deref;
+
+use oxc_ast::{
+    ast::{
+        ArrayExpression, ArrayExpressionElement, CallExpression, Expression, ObjectExpression,
+        ObjectPropertyKind,
+    },
+    AstKind, Visit,
+};
+use oxc_diagnostics::{LabeledSpan, OxcDiagnostic};
+use oxc_macros::declare_oxc_lint;
+use oxc_span::{GetSpan, Span};
+
+use crate::{
+    ast_util::{is_method_call, leftmost_identifier_reference},
+    context::LintContext,
+    rule::Rule,
+    utils::default_true,
+    AstNode,
+};
+
+fn no_map_spread_diagnostic(map_call: Span, spread: &Spread<'_, '_>) -> OxcDiagnostic {
+    let spans = spread.spread_spans();
+    assert!(!spans.is_empty());
+    let mut spread_labels = spread.spread_spans().into_iter(); // .map(LabeledSpan::from);
+    let first_message = if spans.len() == 0 {
```

Comment:
This seems contradictory: we are asserting `spans` is not empty, but then checking if spans length is zero? Do we need to assert here? (If so, could we use `debug_assert!`?)

Reply:
That should be 1, good catch.